iib/web/static/api_v1.yaml

```diff
@@ -0,0 +1,461 @@
+          schema:
+            type: boolean
+            example: true
+            default: false
+      responses:
+        '200':
+          description: A list of build requests
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  items:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Request'
+                  meta:
+                    $ref: '#/components/schemas/Pagination'
+        '400':
+          description: The query parameters are invalid
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    example: >-
+                      awesome is not a valid request state. Valid states are:
+                      complete, failed, in_progress
+        '404':
+          description: No build requests were found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    example: The requested resource was not found
+  '/builds/{id}':
+    get:
+      description: Return a specific build request
+      parameters:
+        - name: id
+          in: path
+          required: true
+          description: The ID of the build request to retrieve
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: The requested build request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RequestVerbose'
+        '404':
+          description: The build request wasn't found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    example: The requested resource was not found
+    patch:
+      description: Update a build request (requires special authorization)
+      parameters:
+        - name: id
+          in: path
+          required: true
+          description: The ID of the build request to retrieve
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: The build request was updated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RequestVerbose'
+        '400':
+          description: The input is invalid
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    example: The input data must be a JSON object
+        '403':
```

Comment:
The API will also return a 401 if the user is not authenticated. Could you please add that?

Reply:
By the looks of the code, it seems like it will return `Forbidden` even if the user is not authenticated.

Reply:
It's confusing, but since authentication is handled by httpd, it will return the 401 with this snippet of the configuration:
```
# Return JSON when authentication fails
ErrorDocument 401 "{\"error\": \"You must be authenticated to perform this action.\"}"
Header always set Content-Type "application/json" "expr=%{REQUEST_STATUS} == 401"
```